File: examples/docs-code-snippets/src/main/scala/examples/scala/visualization/WebShopVisualization.scala
package examples.scala.visualization

import com.ing.baker.compiler.RecipeCompiler
import examples.scala.recipes.WebShopRecipe

class WebShopVisualization {
  def printVisualizationString(): Unit = {
    val compiledRecipe = RecipeCompiler.compileRecipe(WebShopRecipe.recipe)
    val visualization = compiledRecipe.getRecipeVisualization
    println(visualization)
  }
}

File: examples/docs-code-snippets/src/main/java/examples/java/visualization/WebShopVisualization.java
package examples.java.visualization;

import com.ing.baker.compiler.RecipeCompiler;
import com.ing.baker.il.CompiledRecipe;
import examples.java.recipes.WebShopRecipe;

public class WebShopVisualization {
    public void printVisualizationString() {
        CompiledRecipe compiledRecipe = RecipeCompiler.compileRecipe(WebShopRecipe.recipe);
        String graphvizString = compiledRecipe.getRecipeVisualization();
        System.out.println(graphvizString);
    }
}
